source.lua

```diff
@@ -127,18 +181,52 @@ function main(args)
         if not (liquidArg == 'magma' or liquidArg == 'water') then
             qerror('Liquid must be either "water" or "magma"')
         end
-        if not IsFlowPassable(targetPos) then
+        if not is_flow_passable(targetPos) then
             qerror("Tile not flow passable: I'm afraid I can't let you do that, Dave.")
         end
         local amountArg = tonumber(args[3]) or 7
-        AddLiquidSource(targetPos, liquidArg, amountArg)
+        add_liquid_source(targetPos, liquidArg, amountArg)
         print(('Added %s %d at %s'):format(liquidArg, amountArg, formatPos(targetPos)))
         return
     end
+end
 
-    print(dfhack.script_help())
+dfhack.onStateChange[GLOBAL_KEY] = function(sc)
+    if sc ~= SC_MAP_LOADED or df.global.gamemode ~= df.game_mode.DWARF then
+        return
+    end
+    local persisted_data = json.decode(persist.GlobalTable[GLOBAL_KEY] or '') or {}
+    -- sometimes the keys come back as strings; fix that up
+    for k,v in pairs(persisted_data) do
+        if type(k) == 'string' then
+            persisted_data[tonumber(k)] = v
+            persisted_data[k] = nil
+        end
+    end
```

Comment:
you shouldn't have this problem because your keys are list indices -- prioritize stores a map